### ISSUE TITLE
 Omitting normalisation step in `update_PFSMs` has no effect on training tests

### ISSUE THREAD

**Issue by rolyp** [alan-turing-institute/ptype#127]
As discussed, this seems to either be a bug in the training code, or a consequence of poorly-chosen training data. For the training tests to be useful, they have to be sensitive to breaking the training algorithm.

**Comment by tahaceritli** [comment]
See https://github.com/alan-turing-institute/ptype-dmkd/pull/129 for the changes. The training looks fine for the toy example; however, we still have an issue with testing real datasets. Perhaps the issue comes from the datasets used. I will try to understand this.

**Comment by tahaceritli** [comment]
See https://github.com/alan-turing-institute/ptype-dmkd/pull/143 for the inspection of the test datasets. I have summarized the whole inspection in https://github.com/alan-turing-institute/ptype-dmkd/blob/develop/notebooks/train-model.ipynb.
